rewrite-core/src/test/java/org/openrewrite/GitRemoteTest.java

```diff
@@ -102,4 +104,28 @@ void parseRegisteredRemote(String cloneUrl, String origin, GitRemote.Service ser
         assertThat(remote.getOrganization()).isEqualTo(expectedOrganization);
         assertThat(remote.getRepositoryName()).isEqualTo(expectedRepositoryName);
     }
+
+    @ParameterizedTest
+    @CsvSource(textBlock = """
+      GitHub, false, github.com, org/repo, https://github.com/org/repo
+      GitHub, true, github.com, org/repo, ssh://git@github.com/org/repo.git
+      GitHub, false, https://github.com, org/repo, https://github.com/org/repo
+      GitHub, true, https://github.com, org/repo, ssh://git@github.com/org/repo.git
+      GitHub, false, scm.company.com/context/path, org/repo, https://scm.company.com/context/path/org/repo
+      GitHub, true, scm.company.com/context/path, org/repo, ssh://git@scm.company.com/context/path/org/repo.git
+      GitLab, false, gitlab.com, group/subgroup/repo, https://gitlab.com/group/subgroup/repo.git
+      GitLab, true, gitlab.com, group/subgroup/repo, ssh://git@gitlab.com/group/subgroup/repo.git
+      GitLab, false, scm.company.com/context/path, group/subgroup/repo, https://scm.company.com/context/path/group/subgroup/repo.git
+      GitLab, true, scm.company.com/context/path, group/subgroup/repo, ssh://git@scm.company.com/context/path/group/subgroup/repo.git
+      BitbucketCloud, false, bitbucket.org, org/repo, https://bitbucket.org/org/repo
+      BitbucketCloud, true, bitbucket.org, org/repo, ssh://git@bitbucket.org/org/repo.git
+      Bitbucket, false, scm.company.com/context/path, org/repo, https://scm.company.com/context/path/scm/org/repo
+      Bitbucket, true, scm.company.com/context/path, org/repo, ssh://git@scm.company.com/context/path/org/repo.git
```

Comment:
Have we agreed on _not_ adding the port 7999 to the ssh URL for Bitbucket?

Reply:
that is in a separate PR, but we probably need to decide that first: https://github.com/openrewrite/rewrite/pull/4414